test/ranges.jl

```diff
@@ -277,29 +277,49 @@ end
 @test all(((1:5) - [1:5;]) .== 0)
 
 # tricky floating-point ranges
+for (start, step, stop, len) in ((1, 1, 3, 3), (0, 1, 3, 4),
+                                 (3, -1, -1, 5), (1, -1, -3, 5),
+                                 (0, 1, 10, 11), (0, 7, 21, 4),
+                                 (0, 11, 33, 4), (1, 11, 34, 4),
+                                 (0, 13, 39, 4), (1, 13, 40, 4),
+                                 (11, 11, 33, 3), (3, 1, 11, 9),
+                                 (0, 10, 0, 1), (0, -1, 0, 1),
+                                 (0, 10, 55, 0), (0, -1, 5, 0),
+                                 (0, 1, 5, 0))
```

Comment:
previous version of this was equivalent to `(0, 10, 5, 0)` right?

and `(0, -10, 5, 0)` for the second-to-last